pep-9999.rst

```diff
@@ -0,0 +1,204 @@
+PEP: 0999
+Title: Python minimization
+Version: $Revision$
+Last-Modified: $Date$
+Author: 
+Sponsor: 
+Status: Draft
+Type: Standards Track
+Content-Type: text/x-rst
+Created: 15-Sep-2020
+
+
+Abstract
+========
+
+This PEP aims to provide a way for downstream distributors of Python, such as Linux distributions, to minimize the footprint of Python on the hard drive for constrained environments like containers. Two of the possible solutions are provided for discussion.
+
+
+Motivation
+==========
+
+Some downstream Python distributors, e.g. Linux distros, make sure that for every ``.py`` source file is bytecompiled and the corresponding ``.pyc`` file is present on disk. Oftentimes 3 ``.pyc`` bytecache files are present, one non-optimized and one for each optimization level. This has many benefits:
+ - It makes startup times for Python faster.
+ - It sidesteps problems with user privileges, e.g. non-root users usually don't have rights to create files on the root filesystem, possibly generating SELinux warnings in the process (which can be suppressed).
```

Comment:
It's root who generates SELinux denial alerts, not non-root.

Reply:
Wait, there's a "SELinux denial alert" generated when Python run as root tries to write a .pyc file? So the file is not even written when SELinux is in enforcing mode?

And when run as non-root, there's no SELinux message generated?

Reply:
When run as root in restricted context, SELinux will prevent it. When run as regular user, it doesn't work at all and SELinux won't notice/care.

Reply:
Ok, will update.

Reply:
Done. WDYT?

Reply:
resolved